### DISCUSSION TITLE
next/image: implicit size props

### DISCUSSION THREAD

**Discussion by Kikobeats** [vercel/next.js#18309] [Ideas]
# Feature request

## Is your feature request related to a problem? Please describe.

The `next/image` component is so good; however, image size as required props can be a bit tedious to setup per every image in a Next.js project.

> `width` and `height` are required to prevent Cumulative Layout Shift, a Core Web Vital that Google is going to use in their search ranking

I wonder if could be possible to get that information from the image metadata, at least as fallback values

## Describe the solution you'd like

If that information exists, it should be part of the EXIF image metadata information.

<img width="909" alt="Screen Shot 2020-10-27 at 17 58 26" src="https://user-images.githubusercontent.com/2096101/97335335-055f2700-187e-11eb-9a60-cad6c5da9abc.png">

The [exifr](https://github.com/MikeKovarik/exifr) aims for getting that values easily.

Additionally, rather than a Next.js core thing, this could be a helpful tip to include as an example.

**Comment by timneutkens** [comment]
This was explicitly not added for three reasons:
- We'd have to fetch every image in order to read them, this would need extra fetches
- We don't know what url is going to be in the `src` of `next/image`
- We want to pre-size images based on their viewport to avoid layout-shift
